### PR TITLE
Milestone Bubble Sort Project

### PR DESCRIPTION
**Pull request by shubhsk88**
This PR address the implementation of the Buble sort algorithm which was implemented using ruby

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,32 @@
-# bubble-sort-ruby
\ No newline at end of file
+# Bubble sort ruby
+
+This is a project of two algorithms about bubble sorting in ruby
```
Comment (mosesogwo) [inline README.md]: Include a short description of the two algorithms (name and function).

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+def bubble_sort(arr)
+  len = arr.length
+  (0...len).each do |i|
+    (i + 1...len).each do |j|
+      next unless arr[i] > arr[j]
+
+      swap = arr[i]
+      arr[i] = arr[j]
+      arr[j] = swap
```
Comment (mosesogwo) [inline bubble_sort.rb]: You can easily achieve this swap with 
`arr[i], arr[j] = arr[j], arr[i]`

File: bubble_sort_by.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+def bubble_sort_by(arr)
+  len = arr.length
+  (0...len).each do |i|
+    (i + 1...len).each do |j|
+      next unless yield arr[i], arr[j]
+
+      swap = arr[i]
+      arr[i] = arr[j]
+      arr[j] = swap
```
Comment (mosesogwo) [inline bubble_sort_by.rb]: You can easily achieve this swap with 
`arr[i], arr[j] = arr[j], arr[i]`

File: bubble_sort_by.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+def bubble_sort_by(arr)
+  len = arr.length
+  (0...len).each do |i|
+    (i + 1...len).each do |j|
+      next unless yield arr[i], arr[j]
```
Comment (mosesogwo) [inline bubble_sort_by.rb]: This is the point where you should check if the yield is positive, not in the example.
When the yield is positive, it means the left item is greater than the right, and so there should be a swap.

What your code currently does is to swap every time the yield is not false or nil, which will not give the correct output.

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**HI @shubhsk88, @thedekerone:clap:. Good job:+1:. Let's address the following:**
- Your bubble_sort_by is not returning the correct result. See my comment on the method.
- You can move the bubble_sort_by method into bubble_sort.rb so we have a single file.
- See other comments on your project below.


_**Kindly make these changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Review by euqueme — APPROVED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @shubhsk88  and @thedekerone  :balloon: :lollipop: 

### Thank you for doing the changes requested by the previous reviewer, your methods are working fine, please merge this branch and continue congratulations :tada: :tada: :tada: 

[OPTIONAL]

- [ ] I strongly recommend the use of `arr.size` instead of `arr.length` to keep the Ruby convention

PROJECT STATUS: APPROVED :green_apple:

**Pull request merged**